pkg/credentials/config.go

```diff
@@ -99,18 +99,18 @@ func (cp *CredentialsProvider) getExecConfigFromConfig(providerName string) *cli
 	return nil
 }
 
-func (cp *CredentialsProvider) getProviderFromClusterProfile(cluster *v1alpha1.ClusterProfile) *v1alpha1.CredentialProvider {
-	cpProviderTypes := map[string]*v1alpha1.CredentialProvider{}
+func (cp *CredentialsProvider) getClusterAccessFromClusterProfile(cluster *v1alpha1.ClusterProfile) *v1alpha1.AccessProvider {
+	caProviderTypes := map[string]*v1alpha1.AccessProvider{}
 
-	for _, provider := range cluster.Status.CredentialProviders {
+	for _, provider := range cluster.Status.AccessProviders {
 		newProvider := provider.DeepCopy()
-		cpProviderTypes[provider.Name] = newProvider
+		caProviderTypes[provider.Name] = newProvider
 	}
 
 	// we return the first provider that the CP supports.
 	for _, providerType := range cp.Providers {
-		if provider, found := cpProviderTypes[providerType.Name]; found {
-			return provider
+		if accesser, found := caProviderTypes[providerType.Name]; found {
+			return accesser
```

Comment:
We still need to read the old field, so these loops need to process `AccessProviders` and `CredentialProviders`.